docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+
+    const client = new CoffeeShop({ endpoint: { protocol: 'http', hostname: 'localhost', port: 3001, path: '/' } })
+
+    await client.getMenu()
+
+Like when we tested the server with ``cURL``, you should see the output of the ``GetMenu`` operation we implemented.
+Let's try submitting an order:
+
+.. code-block:: TypeScript
+    :caption: ``repl``
+
+    await client.createOrder({ coffeeType: "DRIP" })
+
+After creating the order, you should get response like:
+
+.. code-block:: typescript
+  :caption: response
+
+    {
+      '$metadata': {
+        // metadata, such as response code, added by the client
+      },
+      coffeeType: 'DRIP', // <--- the type of coffee we ordered
+      id: 'ee97e900-d8dd-4770-904c-3d175cda90c3',  // <--- the order id
+      status: 'IN_PROGRESS' // <--- the order status
+    }
+
+The order should be ready by the time you submit this next command. Let's retrieve the order:
+
+.. code-block:: TypeScript
+    :caption: ``repl``
+
+    await client.getOrder({ id: '<PUT YOUR ORDER-ID HERE!>' }) // <--- make sure to replace with your id
+
+Once you execute the command, you should now see your order information:
+
+.. code-block:: typescript
+  :caption: response
+
+    {
+      '$metadata': {
+        // ...
+      },
+      coffeeType: 'DRIP', // <--- the type of coffee we ordered
+      id: 'ee97e900-d8dd-4770-904c-3d175cda90c3',  // <--- the order id
+      status: 'COMPLETED' // <--- the order status, which should be 'COMPLETED'
+    }
+
+With that, you may now terminate the REPL and the server (with ``CTRL + C`` in the respective terminals). We have
+tested each operation that we implemented in the server using the generated client, and verified that both the client
+and server can communicate with each other.
+
+Running the Application
+=======================
+Now that we know how to generate and use a client and server, let's put it all together to use with a web application
+that runs in the browser. The web application exists under the ``app/`` directory, and can be built using the
+``build-app`` make target. The application will run when using the ``run-app`` target. Since this application uses
+the generated client to make requests, the server must be ran alongside the app. For convenience, you may run both
+the web-application and the server in the same terminal:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    make run
+
+While running the application in this way is convenient, it will intertwine the output of the web-app and server. If
+you would like to keep them separate, you should run the other targets (`run-server` and `run-app`). Using the method
+of your choice, launch the server and the application.
+
+.. TODO: Add snippets for how we are calling the service with the client
+
+While this application is incredibly simple, it shows how easily you can integrate a smithy-generated client into an
+application that runs in the browser.
+.. TODO: maybe another sentence on takeaways
```

Comment:
I wonder if we should have takeaway sections for each. Maybe in a admonition (like note, warning, etc) box? Like the review boxes in the quickstart.

Reply:
Hmmm... not a bad idea. I am open to adding that.